### PR TITLE
[API] Move the `zero_infinity` parameter of `CTCLoss` from the forward function to the constructor

### PR DESCRIPTION
**Pull request by aztice**
### PR Category
User Experience

### PR Types
Bug fixes

### Description

将在 #75742 中错误添加到 `paddle.nn.CTCLoss` 中 `forward` 函数参数 `zero_infinity` 移到 `__init__` 上，以保持接口和 PyTorch `torch.nn.CTCLoss`[^1] 一致

#75742 刚刚合入且未发版，尚无需考虑兼容性

[^1]: https://docs.pytorch.org/docs/stable/generated/torch.nn.CTCLoss.html

### CODE REVIEW

File: python/paddle/nn/layer/loss.py [hunk 1]
```diff
@@ -1352,8 +1358,19 @@ def forward(
         input_lengths: Tensor,
         label_lengths: Tensor,
         norm_by_times: bool = False,
-        zero_infinity: bool = False,
+        zero_infinity: bool | None = None,
```
Comment (SigureMo) [inline python/paddle/nn/layer/loss.py]: 为什么还要在这里加这个参数？

Comment (aztice) [reply python/paddle/nn/layer/loss.py]: 我觉得两个入口随便一个其实都行看个人习惯。
del.

Comment (SigureMo) [reply python/paddle/nn/layer/loss.py]: 涉及到 API 不能随便

**Comment by SigureMo** [comment]
描述里说清楚问题，写清上下文

**Comment by aztice** [comment]
> 描述里说清楚问题，写清上下文

ok

File: python/paddle/nn/layer/loss.py (hunk 1 above)
Comment (aztice) [reply python/paddle/nn/layer/loss.py]: > 涉及到 API 不能随便

其实我有点好奇飞桨现在具体的维护方式是什么，感觉用起来乱乱的，除了模型部署和性能上似乎要好一点

**Review by SigureMo — APPROVED** [review]
